app/master/cluster_master.py

```diff
@@ -279,7 +282,7 @@ def handle_result_reported_from_slave(self, slave_url, build_id, subjob_id, payl
         build = self._all_builds_by_id[int(build_id)]
         slave = self._all_slaves_by_url[slave_url]
         # If the build has been canceled, don't work on the next subjob.
-        if not build.is_finished:  # WIP(joey): This check should be internal to the Build object.
+        if not build.is_finished or build.has_error:  # WIP(joey): This check should be internal to the Build object.
```

Comment:
I included this `or build.has_error` logic because it was in the commit from https://github.com/Medium/ClusterRunner/commit/fd38c99155eb0373815b75230f90f09295861770, but I don't see why it's necessary. I tested the change without this line and it still seemed to work fined. It doesn't even seem related to this change.

Do you know why this is here @josephharrington, @tjlee0909? I'm gonna dig for a bit and then get rid of it if I can't find a reason to keep it here.